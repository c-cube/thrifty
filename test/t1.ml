module Fmt = CCFormat

let get_toks, prot = Debug_protocol.debug_write ()

let () =
  let (module P) = prot in
  P.write_struct_begin "Foo";
  P.write_field_begin "x" T_I32 1;
  P.write_i32 42l;
  P.write_field_end ();
  P.write_field_begin "y" T_BOOL 2;
  P.write_bool true;
  P.write_field_end ();
  P.write_struct_end ();
  ()

let () =
  Fmt.printf "tokens: %a@."
    (Fmt.Dump.list Debug_protocol.Token.pp)
    (get_toks ())
