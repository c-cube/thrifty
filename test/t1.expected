tokens: [struct_begin("Foo");field_begin("x",I32,1);42l;field_end;
         field_begin("y",BOOL,2);true;field_end;struct_end]
